protected/humhub/modules/marketplace/Events.php

```diff
@@ -34,50 +43,237 @@ public static function onConsoleApplicationInit($event)
+                'count' => $onlineModulesCount,
+                'view' => '@humhub/modules/marketplace/widgets/views/moduleInstallCard',
+                'sortOrder' => 200,
+            ]);
+        }
     }
 
-    public static function onHourlyCron($event)
+    public static function onAdminModuleManagerAfterFilterModules(ModulesEvent $event)
     {
-        Yii::$app->queue->push(new jobs\PeActiveCheckJob());
-        Yii::$app->queue->push(new jobs\ModuleCleanupsJob());
+        if (!is_array($event->modules)) {
+            return;
+        }
+
+        foreach ($event->modules as $m => $module) {
+            if (!self::isFilteredModule($module)) {
+                unset($event->modules[$m]);
+            }
+        }
+    }
+
+    /**
+     * @param CoreModule|ModelModule $module
+     * @return bool
+     */
+    private static function isFilteredModule($module): bool
+    {
+        return self::isFilteredModuleByCategory($module) &&
+            self::isFilteredModuleByTags($module);
+    }
+
+    /**
+     * @param CoreModule|ModelModule $module
+     * @return bool
+     */
+    private static function isFilteredModuleByCategory($module): bool
+    {
+        $categoryId = Yii::$app->request->get('categoryId', null);
+
+        if (empty($categoryId)) {
+            return true;
+        }
+
+        if (!is_array($module->categories) || empty($module->categories)) {
+            return false;
+        }
+
+        return in_array($categoryId, $module->categories);
+    }
+
+    /**
+     * @param CoreModule|ModelModule $module
+     * @return bool
+     */
+    private static function isFilteredModuleByTags($module): bool
+    {
+        $tags = Yii::$app->request->get('tags', null);
+
+        if (empty($tags)) {
+            return true;
+        }
+
+        $tags = explode(',', $tags);
+
+        foreach ($tags as $tag) {
+            switch ($tag) {
+                case 'installed':
+                    if (!Yii::$app->moduleManager->hasModule($module->id)) {
+                        return false;
+                    }
+                    break;
+                case 'not_installed':
+                    if (Yii::$app->moduleManager->hasModule($module->id)) {
+                        return false;
+                    }
+                    break;
+                case 'professional':
+                    if (!$module->isProOnly()) {
+                        return false;
+                    }
+                    break;
+                case 'featured':
+                    if (!$module->getOnlineInfo('featured')) {
+                        return false;
+                    }
+                    break;
+                case 'official':
+                    if (!$module->getOnlineInfo('isCommunity')) {
+                        return false;
+                    }
+                    break;
+                case 'partner':
+                    if (!$module->getOnlineInfo('isThirdParty')) {
```

Comment:
Show this badge when `isPartner` is `true`. (New Field not avail yet)

Reply:
https://github.com/humhub/humhub/pull/5476/commits/f4163d88cac591e645cc04817f763a0bd42ca509#diff-d29d9ca029353610b8cbd988563fced151f48e64f7b74fe9341335aec477697bR261